lib/query-interface.js

```diff
@@ -495,18 +495,20 @@ QueryInterface.prototype.insert = function(instance, tableName, values, options)
   });
 };
 
-QueryInterface.prototype.upsert = function(tableName, values, updateValues, model, options) {
+QueryInterface.prototype.upsert = function(tableName, valuesByField, updateValues, model, options) {
```

Comment:
Makes sense - thanks for tracking the problem down. 

However, we already have the same code in instance, https://github.com/sequelize/sequelize/blob/68e613050d5e1e57a91c12fdf1b68961a19d70c1/lib/instance.js#L126 - I'll merge the PR and use that instead